src/main/java/com/twilio/taskrouter/domain/common/TwilioAppSettings.java

```diff
@@ -4,113 +4,114 @@
 import com.twilio.sdk.TwilioRestException;
 import com.twilio.sdk.TwilioTaskRouterClient;
 import com.twilio.sdk.resource.instance.Call;
-import com.twilio.sdk.resource.instance.IncomingPhoneNumber;
 import com.twilio.taskrouter.domain.error.TaskRouterException;
 import com.twilio.taskrouter.domain.model.PhoneNumber;
 import org.apache.http.NameValuePair;
 import org.apache.http.message.BasicNameValuePair;
 
 import javax.inject.Singleton;
-import javax.json.Json;
+import java.io.File;
+import java.io.IOException;
 import java.io.UnsupportedEncodingException;
 import java.net.URLEncoder;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
+import java.util.Properties;
+import java.util.logging.Logger;
 
 /**
  * Twilio settings and helper functions for this app
  */
 @Singleton
 public class TwilioAppSettings {
 
-  public static final String DIGITS_PARAM = "Digits";
+  public static final String WORKSPACE_PROPERTIES_FILE_PATH = "workspace.properties";
 
-  public static final String EVENT_TYPE_PARAM = "EventType";
-
-  public static final String TASK_ATTRIBUTES_PARAM = "TaskAttributes";
-
-  public static final List<String> DESIRABLE_EVENTS =
-    Arrays.asList("workflow.timeout", "task.canceled");
+  private static final Logger LOG = Logger.getLogger(TwilioAppSettings.class.getName());
 
   private final TwilioRestClient twilioRestClient;
 
   private final TwilioTaskRouterClient twilioTaskRouterClient;
 
-  private String workFlowSID;
+  private String twilioAccountSid;
 
-  private String postWorkActivitySID;
+  private String twilioAuthToken;
 
-  private String email;
+  private String workflowSid;
+
+  private String workspaceSid;
 
-  private String dequeuInstruction;
+  private String postWorkActivitySid;
 
-  private List<PhoneNumber> activePhoneNumbers;
+  private String email;
+
+  private PhoneNumber phoneNumber;
 
   public TwilioAppSettings() {
-    String twilioAccountSid = Optional.ofNullable(System.getenv("TWILIO_ACCOUNT_SID")).orElseThrow(
-      () -> new TaskRouterException("TWILIO_ACCOUNT_SID is not set in the environment"));
-    String twilioAuthToken = Optional.ofNullable(System.getenv("TWILIO_AUTH_TOKEN")).orElseThrow(
-      () -> new TaskRouterException("TWILIO_AUTH_TOKEN is not set in the environment"));
+    try {
+      Properties properties = Utils.loadProperties(new File(WORKSPACE_PROPERTIES_FILE_PATH));
```

Comment:
You will only read from the environment the first time you use the application and will be the most basic variables, we need the user specify at least once.

* TWILIO_ACCOUNT_SID
* TWILIO_AUTH_TOKEN
* TWILIO_NUMBER
* MISSED_CALLS_EMAIL_ADDRESS

When we create the workspace for the first time, we no longer need to specify these variables again, they are saved together with the new calculated variables into `workspace.properties` which is ignored by git. This gives great advantage, because despite other projects you wont have to specify any variable evermore, just execute the `createWorkspace` in case you want to create again a new workspace.